cblite/llm/OpenAI.cc

```diff
@@ -22,39 +22,16 @@ using namespace std;
 using namespace fleece;
 using namespace litecore;
 
-// LiteCore Request and Response
 alloc_slice OpenAI::run(const string& restBody, C4Error error) {
-    Encoder enc;
-    enc.beginDict();
-    enc["Content-Type"_sl] = "application/json";
-    enc["Content-Length"_sl] = restBody.length();
-    if (getenv("API_KEY") == NULL) {
-        cout << "API Key not provided\n";
-        return 0;
-    }
-    
-    enc["Authorization"] = format("Bearer %s", getenv("API_KEY"));
-    enc.endDict();
-    auto headers = enc.finishDoc();
+    // Run request
+    auto headers = getHeaders(restBody);
     auto r = std::make_unique<REST::Response>("https", "POST", "api.openai.com", 443, "v1/embeddings");
```

Comment:
It looks like the only variable thing between these three calls is the URL and the model name, so I think you can make this class even more refactored in the base class.